### DISCUSSION TITLE
Access global log in Worker thread

### DISCUSSION THREAD

**Discussion by NitramC** [pymeasure/pymeasure#996] [Q&A]
I am having a similar problem to that mentioned in #711. An exception is raised in my Procedure instance, which is handled by handle_error() in the Worker instance. The problem is that I don't know how to get access to the global log referenced in the worker thread -- note this seems to be a different log than that in the main thread of the interactive console I'm running the script in. I would like to add a StreamHandler to that global log instance so that I get a traceback in the console. Any suggestions?

Thanks!

**Comment by NitramC** [comment]
OK, I now understand that in the Worker class
```
global log
log = logging.getLogger()
```
will set log to be the root logger, and that this is accessible from the console. I can then add a StreamHandler to the root logger to emit logs to the console. The issue now is that everything, not just Worker log messages, is streamed.

This works for my current debugging needs, but is using the root logger good practice?

**Reply by BenediktBurger** [reply]
Thanks for raising that issue.

We could change it to `getLogger("Worker")` or something similar or even the typical `getLogger(__name__)`. What do you think @CasperSchippers ?

**Reply by NitramC** [reply]
Good pun haha :)

**Reply by CasperSchippers** [reply]
This is a good point indeed, I think using the root logger is not the most clear thing to do. @BenediktBurger, adding a name seems like a good plan; I'd be inclined to go for `getLogger(__name__)` as not to start using artificial names (which, eventually, will start overlapping if we start doing this more often).

**Reply by CasperSchippers** [reply]
Admittedly, I'm not entirely sure what the purpose is of this construction:
```python
global log
log = logging.getLogger()
```
since `log` is already defined at the beginning of the document, and it seems to work fine when removing those two lines. Then again, removing them seems dangerous, considering Chesterton's Fence.....
